test/framework/src/main/java/org/elasticsearch/test/TaskAssertions.java

```diff
@@ -83,11 +93,31 @@ public static void assertAllCancellableTasksAreCancelled(String actionPrefix) th
         }, 30, TimeUnit.SECONDS);
     }
 
-    public static void assertAllTasksHaveFinished(String actionPrefix) throws Exception {
+    public static void assertAllCancellableTasksAreCancelled(String actionPrefix) throws Exception {
+        assertAllCancellableTasksAreCancelled(actionPrefix, null);
+    }
+
+    private static boolean taskOpaqueIdMatches(Map<String, String> taskHeaders, @Nullable String opaqueId) {
+        if (opaqueId == null) {
```

Comment:
I think we should assert the opaque ID is missing if we didn't expect to see one.